data_structures/bloom_filter.cpp

```diff
@@ -0,0 +1,282 @@
+/**
+ * @file bloom_filter.cpp
+ * @brief [Bloom Filter](https://en.wikipedia.org/wiki/Bloom_filter)
+ * generic implementation in C++
+ * @details A Bloom filter is a space-efficient probabilistic data structure,
+ * a query returns either "possibly in set" or "definitely not in set".
+ *
+ * More generally, fewer than 10 bits per element are required for a 1% false
+ * positive probability, independent of the size or number of elements in the
+ * set.
+ *
+ * It helps us to not make an "expensive operations", like disk IO - we can
+ * use bloom filter to check incoming request, and with a good probability
+ * get an answer of bloom filter, that we don't need to make our "expensive
+ * operation"
+ *
+ * For math details, please, [look
+ * here](https://en.wikipedia.org/wiki/Bloom_filter)
+ *
+ * [Very good use case example](https://stackoverflow.com/a/30247022)
+ *
+ * Basic bloom filter doesn't support deleting of elements, so
+ * we don't need to implement deletion in bloom filter and bitset in our case.
+ * @author [DanArmor](https://github.com/DanArmor)
+ */
+
+#include <cassert>     ///< for assert
+#include <functional>  ///< for list of hash functions for bloom filter constructor
+#include <initializer_list>  ///< for initializer_list for bloom filter constructor
+#include <string>            ///< for testing on strings
+#include <vector>            ///< for vector
+
+/**
+ * @brief Simple bitset implementation for bloom filter
+ */
+class Bitset {
+ private:
+    std::vector<std::size_t> data;  ///< short info of this variable
+    static const std::size_t blockSize =
+        sizeof(std::size_t);  ///< size of integer type, that we are using in
+                              ///< our bitset
+ public:
+    explicit Bitset(std::size_t initSize);
+    std::size_t size();
+    void add(std::size_t x);
+    bool contains(std::size_t x);
+};
+
+/**
+ * @brief Returns size of inner array
+ *
+ * @return size of inner array
+ */
+std::size_t Bitset::size() { return data.size(); }
+
+/**
+ * @brief Constructor for Bitset
+ *
+ * @param initSize amount of blocks, each contain sizeof(std::size_t) bits
+ */
+Bitset::Bitset(std::size_t initSize) : data(initSize) {}
+
+/**
+ * @brief Turn bit on position x to 1s
+ *
+ * @param x position to turn bit on
+ * @return void
+ */
+void Bitset::add(std::size_t x) {
+    std::size_t blockIndex = x / blockSize;
+    if (blockIndex >= data.size()) {
+        data.resize(blockIndex + 1);
+    }
+    data[blockIndex] |= 1 << (x % blockSize);
+}
+
+/**
+ * @brief Doest bitset contains element x
+ *
+ * @param x position in bitset to check
+ * @return true if: bit on position x is 1
+ * @return false if: bit on position x is 0
```

Comment:
Doxygen will adjust the newlines here. Without the dots, it will look better.
```suggestion
 * @return true if bit on position x is 1
 * @return false if bit on position x is 0
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix